src/ng-token-auth.coffee

```diff
@@ -387,26 +388,48 @@ angular.module('ng-token-auth', ['ipCookie'])
             @cancelOmniauthInAppBrowserListeners()
             handleAuthWindowClose = @handleAuthWindowClose.bind(this, authWindow)
             handleLoadStop = @handleLoadStop.bind(this, authWindow)
+            handlePostMessage = @handlePostMessage.bind(this)
 
             authWindow.addEventListener('loadstop', handleLoadStop)
             authWindow.addEventListener('exit', handleAuthWindowClose)
+            authWindow.addEventListener('message', handlePostMessage)
 
             this.cancelOmniauthInAppBrowserListeners = () ->
               authWindow.removeEventListener('loadstop', handleLoadStop)
               authWindow.removeEventListener('exit', handleAuthWindowClose)
+              authWindow.addEventListener('message', handlePostMessage)
 
 
           # responds to inAppBrowser window loads
           handleLoadStop: (authWindow) ->
             _this = this
-            authWindow.executeScript({code: 'requestCredentials()'}, (response) ->
+
+            # favor InAppBrowser postMessage API if available, otherwise revert to returning directly via
+            # the executeScript API, which is known to have limitations on payload size
+            remoteCode = "
+              function performBestTransit() {
```

Comment:
Backwards-compatible change with DTA InAppBrowser handling. Check to see if the new `authWindow` implementation supports the `cordova-in-app-browser`'s `postMessage` implementation, and if so, prefer that. `executeScript` has some limitations in payload size that may otherwise cause silent failures.

If the `postMessage` API is available, we will utilize our existing callback handler to retrieve the `data`.

If the `postMessage` API implementation is not available, default to returning the `data` directly through `executeScript`, as it did previously.